contrib/how-to-build.md

```diff
@@ -0,0 +1,56 @@
+# How To Build With `cmake`
+
+To build this project you can still use `npm install`, but here some advice to improve your development experience:
+
+
+## Getting Started
+
+First clone the project:
+```sh
+git clone git@github.com:realm/realm-js.git
+
+# The only branch supporting cmake at the moment is develop
+git checkout develop
+```
+
+Pull the submodules:
+
+```sh
+git submodule update --init --recursive
+```
+> If this folder `src/object-store` exist then do a `git clean -xfd`.
+
+
+
+## Configuration
+Now you should get [cmake-js](https://www.npmjs.com/package/cmake-js) which is a [node-gyp](https://github.com/nodejs/node-gyp) replacement:
+
+```sh
+  npm install cmake-js -g
```

Comment:
In general I would advice against globally installed packages, since these are hard to keep updated and we would be running different versions locally and on CI.

Instead use dev-dependencies (as Kenneth mentions, it is a dev-dependency now) and the `npx cmake-js` command to run its CLI.

```suggestion
  npm install
```